mock-plugin/src/test/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImplTest.java

```diff
@@ -0,0 +1,272 @@
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedData);
+        ResponseWrapper<JsonNode> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentity);
+        Error error = new Error();
+        error.setErrorCode("invalid_individual_id");
+        responseWrapper.setErrors(List.of(error));
+        ResponseEntity<ResponseWrapper<JsonNode>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+
+        Mockito.when(restTemplate.exchange(
+                "http://localhost:8080/"+individualId,
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<ResponseWrapper<JsonNode>>() {
+                })).thenReturn(responseEntity);
+
+        ProfileDto profileDto= mockProfileRegistryPlugin.getProfile(individualId);
+        Assert.assertNotNull(profileDto);
+        Assert.assertEquals(profileDto.getIndividualId(),individualId);
+    }
+
+
+    @Test
+    public void updateProfile_withVerifiedClaim_thenPass()  {
+
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "objectMapper",objectMapper);
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "addVerifiedClaimsEndpoint","http://localhost:8080/");
+        String requestId = "req-123";
+        String individualId = "ind-456";
+
+        Map<String, Object> verifiedData = new HashMap<>();
+        verifiedData.put("email","123@email.com");
+        verifiedData.put("password","123456");
+
+        Map<String,Object> verifiedClaim=new HashMap<>();
+        verifiedClaim.put("verified_claims",verifiedData);
+
+        JsonNode mockIdentity = objectMapper.valueToTree(verifiedClaim);
+        ProfileDto profileDto = new ProfileDto();
+        profileDto.setIndividualId(individualId);
+        profileDto.setIdentity(mockIdentity);
+
+
+        MockIdentityResponse mockIdentityResponse=new MockIdentityResponse();
+        mockIdentityResponse.setStatus("UPDATED");
+        ResponseWrapper<MockIdentityResponse> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setResponse(mockIdentityResponse);
+        ResponseEntity<ResponseWrapper<MockIdentityResponse>> responseEntity=new ResponseEntity<>(responseWrapper, HttpStatus.OK);
+        Mockito.when(restTemplate.exchange(
+                Mockito.anyString(),
+                Mockito.any(HttpMethod.class),
+                Mockito.any(),
+                Mockito.eq(new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {
+                }))).thenReturn(responseEntity);
+
+
+        ProfileResult profileResult = mockProfileRegistryPlugin.updateProfile(requestId, profileDto);
+        Assert.assertNotNull(profileResult);
+        Assert.assertEquals(profileResult.getStatus(),"UPDATED");
+    }
+
+    @Test
+    public void updateProfile_withOutVerifiedClaim_thenPass()  {
+
+        ReflectionTestUtils.setField(mockProfileRegistryPlugin, "objectMapper",objectMapper);
```

Comment:
Why is this declared in each test case, can you define this in "@Before" method?

Reply:
moved it two common method